enhancements/machine-config/mco-network-configuration.md

```diff
@@ -0,0 +1,253 @@
+---
+title: mco-network-configuration
+authors:
+  - "@bcrochet"
+reviewers:
+  - "@cwalters"
+approvers:
+  - TBD
+creation-date: 2020-06-25
+last-updated: 2020-06-25
+status: provisional
+---
+
+# Extend MCO to support declarative host network configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The ability to configure additional host networking parameters is paramount on some platforms, especially baremetal. Baremetal nodes usally have many interfaces that need to be dynamically configured to access other, critical datacenter resources. Doing so with MachineConfig would incur reboots that are expensive in time on the baremetal platform. The ability to configure host networking parameters without rebooting is desirable. It is also desirable to rollback automatically on failure, as opposed to failing on reboot.
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+
+The [Container Network Interface](https://github.com/containernetworking/cni) (CNI) standard enables different solutions for connecting networks on the node with pods. Some of them are [part of the standard](https://github.com/containernetworking/plugins), and there are others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni), [SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster, with dynamic networking requirements, is a challenge by itself - and this is what this project is addressing.
+
+The node network configuration shall remain in the MachineConfigOperator workflow, but be declarative.
```

Comment:
Could you please elaborate about the flow of  day-1 configuration? 

Will it be possible to use the same flow for day-1 and day-2 networking operations?

Reply:
We are not necessarily targeting day 1 with this proposal at this time.